src/common/log.rs

```diff
@@ -0,0 +1,40 @@
+use crate::google::protobuf::{value::Kind, Struct, Timestamp, Value};
+use crate::proto::app::v1::LogEntry;
+use chrono::{DateTime, FixedOffset};
+use std::collections::HashMap;
+
+pub fn config_log_entry(
+    cfg_received_datetime: DateTime<FixedOffset>,
+    err: Option<&anyhow::Error>,
+) -> LogEntry {
+    let secs = cfg_received_datetime.timestamp();
+    let nanos = cfg_received_datetime.timestamp_subsec_nanos();
+    let level = match err {
+        Some(_) => "error".to_string(),
+        None => "info".to_string(),
+    };
+    let message = match err {
+        Some(err) => format!("could not create robot from config: {err}"),
```

Comment:
looks like we could pass it as a generic implementing the display trait

Reply:
I'm not sure I understand, do you mean the message should be the argument to the function instead of an error and that argument should be type `dyn std::fmt::Display`?